pkg/scalers/nats_jetstream_scaler.go

```diff
@@ -149,11 +152,24 @@ func parseNATSJetStreamMetadata(config *ScalerConfig) (natsJetStreamMetadata, er
 	}
 
 	meta.scalerIndex = config.ScalerIndex
+
+	meta.useHTTPS = false
+	if val, ok := config.TriggerMetadata["useHTTPS"]; ok {
+		useHTTPS, err := strconv.ParseBool(val)
+		if err != nil {
+			return meta, fmt.Errorf("useHTTPS parsing error %s", err.Error())
+		}
+		meta.useHTTPS = useHTTPS
+	}
 	return meta, nil
 }
 
 func (s *natsJetStreamScaler) getNATSJetStreamEndpoint() string {
-	return fmt.Sprintf("http://%s/jsz?acc=%s&consumers=true&config=true", s.metadata.monitoringEndpoint, s.metadata.account)
+	protocol := natsHTTPProtocol
+	if s.metadata.useHTTPS {
+		protocol = natsHTTPSProtocol
+	}
```

Comment:
This function is executed on every metric request, WDYT about moving this logic to the parsing function to improve the code interpolating this string only once?

Reply:
Do you mean to move the function `getMonitoringEndpoint()` to the parsing function and add the complete endpoint to `natsJetStreamMetadata`?

Reply:
Yeah,
Or you could also remove this function and place these lines directly in the parsing function but at least, the endpoint should be evaluated in the parsing function and store as a property to evaluate it only once instead of on every request